### PR TITLE
Clean-up YANG groupings and components identifiers

### LINKED ISSUE #37: Clean-up YANG groupings
> Additional comments that may be addressed after the PR is merged (if not addressed, we can track them as new open issues):
> 1. There is no need for empty grouping for the component-specific attributes. An empty container defined as an augmentation target should be enough for the base network inventory model
> 2. It is preferable to avoid defining grouping when used once unless this would help the readability of the code
> 

_Originally posted by @italobusi in https://github.com/ietf-ivy-wg/network-inventory-yang/pull/34#pullrequestreview-1865791290_

### LINKED ISSUE #57: Align with rfc8348
https://github.com/ietf-ivy-wg/network-inventory-yang/blob/d13257a88ddef39df709dd6dd803c9a9c3361302/ietf-network-inventory.yang#L285

Can this be aligned with rfc8348 naming?

### PR DESCRIPTION
**Pull request by italobusi**
Cleaned-up use of YANG groupings: fix #37
Added text clarifying the design choice for component identifier: fix #57 

---

Co-Authored-By: YuChaode <yuchaode@huawei.com>

### CODE REVIEW

File: draft-ietf-ivy-network-inventory-yang.md [hunk 1]
```diff
@@ -475,7 +475,7 @@ The hierarchical components' identifier could be found by the "component-referen
 
 ### Component-Specific Info Design
 
-According to the management requirements from operators, some important attributes are not defined in {{!RFC8348}}. These attributes could be component-specific and are not suitable to define under the component list node. So, the model can be augmented with HW-specific-info grouping containing attributes applicable to HW e.g. boards/slot components only. Other component-specific attributes, such as SW-specific-info, may be defined in companion augmentation data models, such as
+According to the management requirements from operators, some important attributes are not defined in {{!RFC8348}}. These attributes could be component-specific and are not suitable to define under the component list node. So, component-specific attributes can be defined by augmenting the component-specific info containers for the attributes applicable to HW e.g. boards/slot components only. Other component-specific attributes, such as SW-specific-info, may be defined in companion augmentation data models, such as
```
Comment (aguoietf) [inline draft-ietf-ivy-network-inventory-yang.md:478]: %s/not suitable to define/not suitable to be defined

%s/So, ... components only/ Instead, they can be defined by augmenting the component-specific info container for the attributes applicable to HW...only.

File: ietf-network-inventory.yang [hunk 2]
```diff
@@ -90,26 +94,31 @@ module ietf-network-inventory {
...
-      "A set of attributes which are common to all the entities
-       (e.g., component, equipment room) defined in this module.";
+      "The set of attributes which are common to all the entities
+       (e.g., component, network elements) defined in this module.";
     leaf uuid {
       type yang:uuid;
       config false;
       description
-        "Uniquely identifies an entity (e.g., component).";
+        "The Universallu Unique Identifier of the entity
```
Comment (aguoietf) [inline ietf-network-inventory.yang:111]: %s/Universallu/Universally

**Review by aguoietf — APPROVED** [review]

File: ietf-network-inventory.yang (hunk 2 above)
Comment (italobusi) [reply ietf-network-inventory.yang:111]:
```suggestion
        "The Universally Unique Identifier of the entity
```

File: draft-ietf-ivy-network-inventory-yang.md (hunk 1 above)
Comment (italobusi) [reply draft-ietf-ivy-network-inventory-yang.md:449]:
```suggestion
According to the management requirements from operators, some important attributes are not defined in {{!RFC8348}}. These attributes could be component-specific and are not suitable to be defined under the component list node. Instead, they can be defined by augmenting the component-specific info container for the attributes applicable to HW e.g. boards/slot components only. Other component-specific attributes, such as SW-specific-info, may be defined in companion augmentation data models, such as
```

File: ietf-network-inventory.yang [hunk 3]
```diff
@@ -116,134 +126,75 @@ module ietf-network-inventory {
...
     leaf serial-number {
       type string;
       config false;
       description
-        "The vendor-specific serial number of the network element
-         instance. It is expected that vendors assign unique serial
-         numbers to different network element instances within the
-         scope of the product name.";
+        "The vendor-specific serial number of the the entity
+         (e.g., component) instace. It is expected that vendors
```
Comment (sergiobelotti) [inline ietf-network-inventory.yang:176]: c/instace/instance

**Review by sergiobelotti — APPROVED** [review]

File: ietf-network-inventory.yang (hunk 3 above)
Comment (italobusi) [reply ietf-network-inventory.yang:176]:
```suggestion
         (e.g., component) instance. It is expected that vendors
```

**Review by prmanna — APPROVED** [review]

**Pull request merged**